pkg/image/docker/tag.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/heptio/sonobuoy/pkg/image/exec"
+)
+
+// Tag tags an image, retrying up to retries times
+func Tag(src, dest string, retries int) error {
+	log.Infof("Tagging image: %s as %s ...", src, dest)
```

Comment:
An optimization but if the src==dest we should early return. Those are effectively images they dont want to retag/push.

If it makes more sense then maybe it should be an error (or a special error/type) so that `push` knows not to try and push that since it didnt get retagged?

```
INFO[0043] Tagging image: gcr.io/google-samples/gb-frontend:v6 as gcr.io/google-samples/gb-frontend:v6 ...
```

Reply:
I worry about erroring out here too much. There are so many images you end up having to do alot of manual effort to get past the errors. If you retag the same image, it's essentially a noop on the local client.

Reply:
Sorry if I wasnt clear; what I was saying that this _function_ could return a sentinel error (or error type/interface) so that it exits early and doesnt shell out to docker. Then the calling function would handle it appropriately.

E.g.
```
func tagAll(images []string){
  for _,i:=range images{
    err := tag(src,dest,retries)
    switch err{
    case err==sameTagErr:
      log.Infof("Source and destination tags match; skipping...")
...
...
```

Shelling out to the local docker agent and retagging the image is a noop so thats not that big a deal; the messaging to the user "Tagging X as X" is just a little weird. 

The bigger issue is when you are then trying to push right after this: in 99.9% of all cases the user doesn't have authority to push to the default registry and it will fail. If we _know_ they aren't retagging image X, why go through the process of tagging X as X, pushing X then failing, retrying to push X, then erroring when all our retries fail.

In that flow we end up with 5 log lines for something we _knew_ was a failure (noop at best)